src/data_storage/graph/node_storage.ts

```diff
@@ -189,21 +193,67 @@ export default class NodeStorage extends PostgresStorage{
         return super.run(NodeStorage.archiveStatement(id))
     }
 
-    private async validateNodeProperties(typeKeys: MetatypeKeyT[], input:any): Promise<Result<boolean>> {
+    private async validateAndTransformNodeProperties(typeKeys: MetatypeKeyT[], input:any): Promise<Result<any>> {
         // easiest way to create type for callback func
         const compiledType = CompileMetatypeKeys(typeKeys);
 
-        // allows us to accept an array of input if needed
-        const payload = (t.array(t.unknown).is(input)) ? input : [input];
+
+        // before we attempt to validate we need to insure that any keys with default values have that applied to the payload
+        for(const key of typeKeys) {
+            if(key.property_name in input) continue;
+
+            switch(key.data_type) {
+                case "number": {
+                    input[key.property_name] = +key.default_value!
+                    break;
+                }
+
+                case "boolean": {
+                    input[key.property_name] = key.default_value === "true" || key.default_value === "t"
```

Comment:
Does this check mean that only a boolean value with a default of true/t will be handled correctly?

Reply:
Anything not `true` or `t` as the default value for a boolean will be treated as `false`